gettsim/transfers/grundrente.py

```diff
@@ -0,0 +1,397 @@
+    )
+    return out
+
+
+def _durchschnittl_entgeltp_g_r_bewertungszeiten(
+    entgeltp_grundr: FloatSeries, g_r_bewertungszeiten: IntSeries
+) -> FloatSeries:
+    """Compute average number of Entgeltpunkte earned per month of
+    Grundrentenbewertungszeiten.
+
+    Parameters
+    ----------
+    entgeltp_grundr
+        See basic input variable
+        :ref:`entgeltp_grundr <entgeltp_grundr>`.
+    g_r_bewertungszeiten
+        See basic input variable
+        :ref:`g_r_bewertungszeiten <g_r_bewertungszeiten>`.
+
+    Returns
+    -------
+
+    """
+
+    return entgeltp_grundr / g_r_bewertungszeiten
+
+
+@add_rounding_spec(params_key="ges_rentenv")
+def höchstwert_grundr_zuschlag_m(
+    grundrentenzeiten: IntSeries, ges_rentenv_params: dict
+) -> FloatSeries:
+    """Calculates the maximum allowed number of average Entgeltpunkte (per month)
+    after adding bonus of Entgeltpunkte for a given number of Grundrentenzeiten.
+
+    Parameters
+    ----------
+    grundrentenzeiten
+        See basic input variable :ref:`grundrentenzeiten <grundrentenzeiten>`.
+    ges_rentenv_params
+        See params documentation :ref:`ges_rentenv_params <ges_rentenv_params>`.
+
+    Returns
+    -------
+
+    """
+
+    # Calculate number of months above minimum threshold
+    months_above_thresh = (
+        grundrentenzeiten.clip(upper=ges_rentenv_params["grundrentenzeiten"]["max"])
+        - ges_rentenv_params["grundrentenzeiten"]["min"]
+    )
+
+    # Calculate höchstwert
+    out = (
+        ges_rentenv_params["höchstwert_grundrente"]["base"]
+        + ges_rentenv_params["höchstwert_grundrente"]["increment"] * months_above_thresh
+    )
+
+    return out
+
+
+@add_rounding_spec(params_key="ges_rentenv")
+def grundr_zuschlag_bonus_entgeltp(
+    _durchschnittl_entgeltp_g_r_bewertungszeiten: FloatSeries,
+    höchstwert_grundr_zuschlag_m: FloatSeries,
+    grundrentenzeiten: IntSeries,
+    ges_rentenv_params: dict,
+) -> FloatSeries:
+    """Calculate additional Entgeltpunkte for pensioner.
+
+    In general, the average of monthly Entgeltpunkte earnd in Grundrentenzeiten is
+    doubled, or extended to the individual Höchstwert if doubling would exceed the
+    Höchstwert. Then, the value is multiplied by 0.875.
+
+    Legal reference: § 76g SGB VI
+
+    Parameters
+    ----------
+    _durchschnittl_entgeltp_g_r_bewertungszeiten
+        See :func:`_durchschnittl_entgeltp_g_r_bewertungszeiten`.
+    höchstwert_grundr_zuschlag_m
+        See :func:`höchstwert_grundr_zuschlag_m`.
+    grundrentenzeiten
+        See basic input variable :ref:`grundrentenzeiten <grundrentenzeiten>`.
+    ges_rentenv_params
+        See params documentation :ref:`ges_rentenv_params <ges_rentenv_params>`.
+
+    Returns
+    -------
+
+    """
+    out = pd.Series(0, index=grundrentenzeiten.index)
+
+    # Case 1: Entgeltpunkte less than half of Höchstwert
+    below_half_höchstwert = _durchschnittl_entgeltp_g_r_bewertungszeiten <= (
```

Comment:
`piecewise_polynomial` currently only allows for thesholds that are constant for all subjects. But I agree that this would be easier. Is it ok if I make a comment to revise when moving to scalars and to improve piecewise_polynomial in the process?

Reply:
Sure.